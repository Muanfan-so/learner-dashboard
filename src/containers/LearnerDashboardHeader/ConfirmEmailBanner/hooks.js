import React from 'react';

import { StrictDict } from 'utils';
import { hooks as appHooks, thunkActions } from 'data/redux';

import { useDispatch } from 'react-redux';

import * as module from './hooks';

export const state = StrictDict({
  showPageBanner: (val) => React.useState(val), // eslint-disable-line
  showConfirmModal: (val) => React.useState(val), // eslint-disable-line
});

export const useConfirmEmailBannerData = () => {
  const dispatch = useDispatch();
  const { isNeeded } = appHooks.useEmailConfirmationData();
  const [showPageBanner, setShowPageBanner] = module.state.showPageBanner(isNeeded);
  const [showConfirmModal, setShowConfirmModal] = module.state.showConfirmModal(false);
  const closePageBanner = () => setShowPageBanner(false);
  const closeConfirmModal = () => setShowConfirmModal(false);
  const openConfirmModal = () => setShowConfirmModal(true);

  const openConfirmModalButtonClick = () => {
    dispatch(thunkActions.app.sendConfirmEmail());
    openConfirmModal();
  };

  const userConfirmEmailButtonClick = () => {
    closeConfirmModal();
    closePageBanner();
  };
  return {
    isNeeded,
    showPageBanner,
    closePageBanner,
    showConfirmModal,
    closeConfirmModal,
    openConfirmModalButtonClick,
    userConfirmEmailButtonClick,
  };
};

export default useConfirmEmailBannerData;
