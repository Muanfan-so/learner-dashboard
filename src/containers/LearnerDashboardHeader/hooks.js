import React from 'react';
import { useWindowSize, breakpoints } from '@edx/paragon';
import track from 'tracking';

export const useIsCollapsed = () => {
  const { width } = useWindowSize();
  const isCollapsed = React.useMemo(() => (width <= breakpoints.large.maxWidth), [width]);
  return isCollapsed;
};

export const findCoursesNavClicked = (href) => track.findCourses.findCoursesClicked(href, {
  linkName: 'learner_home_nav_discover',
});

export const findCoursesNavDropdownClicked = (href) => track.findCourses.findCoursesClicked(href, {
  linkName: 'learner_home_nav_dropdown_discover',
});

export default {
  useIsCollapsed,
  findCoursesNavClicked,
  findCoursesNavDropdownClicked,
};
