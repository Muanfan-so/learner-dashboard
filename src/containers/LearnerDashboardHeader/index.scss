// Seems to not be used anywhere, can this be removed?
.dropdown-menu-collapse {
  width: 100vw;
  position: absolute;
  left: 0;
}

// Hard coded border bottom to indicate which tab we are on. The transparent border bottom is to align all the navigation links in the headers. Seems like a similar affect could be accomplishged using margin.
.learner-variant-header {
  a {
    // Hard coded, not sure of the best way to address this
    border-bottom: 2px solid transparent;
  }
  .course-link {
    border-bottom: 2px solid !important;
  }

  .course-link:hover {
    border-bottom: inherit !important;
  }
}

// Hard coded many items in the nav collapsed menu, would be hard to style
.nav-small-menu {
  > * {
    justify-content: flex-start !important; 

    border-radius: 0 !important; // Could potentially be replaced by paragon's styles?
    border-top: 1px solid #ddd !important; // Could potentially be replaced by paragon's styles?

    &::after {
      content: '\00BB';
      padding-left: 10px; // Could potentially be replaced by paragon's styles?
    }
  }
}

.logo {
  // Removed hard coded px height. Suggested change comes from frontend component header in order to have some consistencies. It seems to work just fine when using the same exact style that we have for logo in frontend component header
  display: block;
    box-sizing: content-box;
    position: relative;
    top: -.05em;
    height: 1.75rem;
    padding: 1rem 0;
    margin-right: 1rem;
    img {
      display: block;
      height: 100%;
    }
}
