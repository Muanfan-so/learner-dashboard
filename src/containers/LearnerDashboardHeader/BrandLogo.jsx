import React from 'react';

import { useIntl } from '@edx/frontend-platform/i18n';

import { reduxHooks } from 'hooks';
import { configuration } from '../../config';

import messages from './messages';

export const BrandLogo = () => {
  const { formatMessage } = useIntl();
  const dashboard = reduxHooks.useEnterpriseDashboardData();

  return (
    <a href={dashboard?.url || '/'} className="mx-auto">
      <img
        className="logo py-3"
        src={configuration.LOGO_URL}
        alt={formatMessage(messages.logoAltText)}
      />
    </a>
  );
};

BrandLogo.propTypes = {};

export default BrandLogo;
