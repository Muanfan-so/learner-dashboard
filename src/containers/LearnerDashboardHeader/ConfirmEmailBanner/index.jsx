/* eslint-disable max-len */
import React from 'react';
import {
  PageBanner, ModalDialog, MarketingModal, Button,
} from '@edx/paragon';
import { useIntl } from '@edx/frontend-platform/i18n';

import { hooks as appHooks } from 'data/redux';
import messages from './messages';
import './ConfirmEmailBanner.scss';

export const ConfirmEmailBanner = () => {
  const { isNeeded } = appHooks.useEmailConfirmationData();
  const [show, setShow] = React.useState(isNeeded);
  const [isOpen, setOpen] = React.useState(false);
  const close = () => setOpen(false);
  const { formatMessage } = useIntl();

  if (!isNeeded) { return null; }

  return (
    <>
      <PageBanner show={show} dismissible onDismiss={() => setShow(false)}>
        {formatMessage(messages.confirmEmailTextReminderBanner, {
          confirmNowButton: (
            <Button
              className="confirm-email-now-button"
              variant="link"
              size="inline"
              onClick={() => {
                setOpen(true);
                setShow(false);
              }}
            >
              {formatMessage(messages.confirmNowButton)}
            </Button>
          ),
        })}
      </PageBanner>
      <MarketingModal
        title="My dialog"
        isOpen={isOpen}
        onClose={close}
        hasCloseButton={false}
        heroNode={(
          <ModalDialog.Hero className="bg-gray-300">
            <img className="m-auto" src="confirm-email.svg" alt="" />
          </ModalDialog.Hero>
        )}
        footerNode={(
          <Button className="mx-auto my-3" variant="danger" onClick={close}>
            {formatMessage(messages.verifiedConfirmEmailButton)}
          </Button>
        )}
      >
        <h1 className="text-center p-3">{formatMessage(messages.confirmEmailModalHeader)}</h1>
        <p className="text-center">{formatMessage(messages.confirmEmailModalBody)}</p>
      </MarketingModal>
    </>
  );
};
ConfirmEmailBanner.propTypes = {};

export default ConfirmEmailBanner;
