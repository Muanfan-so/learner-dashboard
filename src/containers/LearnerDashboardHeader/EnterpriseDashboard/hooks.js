import React from 'react';
import { hooks as appHooks } from 'data/redux';
import { StrictDict } from 'utils';
import * as module from './hooks';

export const state = StrictDict({
  showDialog: (val) => React.useState(val), // eslint-disable-line
  selectedItem: (val) => React.useState(val), // eslint-disable-line
});

export const useEnterpriseDashboardHook = () => {
  const { availableDashboards, mostRecentDashboard } = appHooks.useEnterpriseDashboardData();
  const [showDialog, setShowDialog] = module.state.showDialog(false);
  const [selectedItem, setSelectedItem] = module.state.selectedItem({});

  const beginSelectDashboardItem = (val) => () => {
    setSelectedItem(val);
    setShowDialog(true);
  };

  const cancelSelectDashboardItem = () => {
    setSelectedItem({});
    setShowDialog(false);
  };

  return {
    availableDashboards,
    mostRecentDashboard,
    showDialog,

    selectedItem,
    beginSelectDashboardItem,
    cancelSelectDashboardItem,
  };
};

export default useEnterpriseDashboardHook;
