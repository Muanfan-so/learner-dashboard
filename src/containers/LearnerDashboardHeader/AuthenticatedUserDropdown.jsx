import React from 'react';
import PropTypes from 'prop-types';

import { getConfig } from '@edx/frontend-platform';
import { useIntl } from '@edx/frontend-platform/i18n';
import { Dropdown, Icon } from '@edx/paragon';
import { Person } from '@edx/paragon/icons';

import messages from './messages';
import EnterpriseDashboard from './EnterpriseDashboard';

export const AuthenticatedUserDropdown = ({ username }) => {
  const { formatMessage } = useIntl();
  return (
    <>
      <Dropdown className="user-dropdown">
        <Dropdown.Toggle id="user" variant="primary">
          <Icon src={Person} />
          <span data-hj-suppress className="d-none d-md-inline">
            {username}
          </span>
        </Dropdown.Toggle>
        <Dropdown.Menu className="dropdown-menu-right">
          <EnterpriseDashboard />
          <Dropdown.Item href={`${getConfig().LMS_BASE_URL}/u/${username}`}>
            {formatMessage(messages.profile)}
          </Dropdown.Item>
          <Dropdown.Item href={`${getConfig().LMS_BASE_URL}/account/settings`}>
            {formatMessage(messages.account)}
          </Dropdown.Item>
          {getConfig().ORDER_HISTORY_URL && (
            <Dropdown.Item href={getConfig().ORDER_HISTORY_URL}>
              {formatMessage(messages.orderHistory)}
            </Dropdown.Item>
          )}
          <Dropdown.Item href={getConfig().SUPPORT_URL}>
            {formatMessage(messages.help)}
          </Dropdown.Item>
          <Dropdown.Item href={getConfig().LOGOUT_URL}>
            {formatMessage(messages.signOut)}
          </Dropdown.Item>
        </Dropdown.Menu>
      </Dropdown>
    </>
  );
};

AuthenticatedUserDropdown.propTypes = {
  username: PropTypes.string.isRequired,
};

export default AuthenticatedUserDropdown;
