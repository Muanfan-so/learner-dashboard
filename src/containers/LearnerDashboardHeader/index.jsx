import React, { useContext } from 'react';

import { useIntl } from '@edx/frontend-platform/i18n';
import { AppContext } from '@edx/frontend-platform/react';
import { Program } from '@edx/paragon/icons';
import { Button, Image } from '@edx/paragon';

import topBanner from 'assets/top_stripe.svg';
import MasqueradeBar from 'containers/MasqueradeBar';
import urls from 'data/services/lms/urls';

import AuthenticatedUserDropdown from './AuthenticatedUserDropdown';
import GreetingBanner from './GreetingBanner';
import ConfirmEmailBanner from './ConfirmEmailBanner';

import { useIsCollapsed } from './hooks';
import messages from './messages';
import './index.scss';

const { programsUrl } = urls;

export const UserMenu = () => {
  const { authenticatedUser } = useContext(AppContext);
  return authenticatedUser ? (<AuthenticatedUserDropdown username={authenticatedUser.username} />) : null;
};

export const LearnerDashboardHeader = () => {
  const { formatMessage } = useIntl();
  const isCollapsed = useIsCollapsed();

  return (
    <>
      <ConfirmEmailBanner />
      <div className="flex-column bg-primary">
        {!(isCollapsed) && (
          <Image className="d-block w-100" src={topBanner} />
        )}
        <header className="learner-dashboard-header">
          <div className="d-flex">
            {(!isCollapsed) && (
              <Button as="a" href={programsUrl} variant="inverse-tertiary" iconBefore={Program}>
                {formatMessage(messages.switchToProgram)}
              </Button>
            )}
            <div className="flex-grow-1">
              {isCollapsed && <GreetingBanner size="small" />}
            </div>
            {isCollapsed
              ? (<div className="my-auto ml-1"><UserMenu /></div>)
              : (<UserMenu />)}
          </div>
        </header>
        {!isCollapsed && <GreetingBanner size="large" />}
      </div>
      <MasqueradeBar />
    </>
  );
};

LearnerDashboardHeader.propTypes = {
};

export default LearnerDashboardHeader;
