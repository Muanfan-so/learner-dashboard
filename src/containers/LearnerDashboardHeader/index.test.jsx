import { shallow } from 'enzyme';
import { AppContext } from '@edx/frontend-platform/react';

import LearnerDashboardHeader, { UserMenu } from '.';

import { useIsCollapsed } from './hooks';

jest.mock('@edx/frontend-platform/react', () => ({
  AppContext: {
    authenticatedUser: {
      username: 'test-username',
    },
  },
}));
jest.mock('./hooks', () => ({
  useIsCollapsed: jest.fn(),
  findCoursesNavClicked: (href) => jest.fn().mockName(`findCoursesNavClicked(${href})`),
}));
jest.mock('data/redux', () => ({
  hooks: {
    usePlatformSettingsData: jest.fn(() => ({
      courseSearchUrl: 'test-course-search-url',
    })),
  },
}));
jest.mock('containers/MasqueradeBar', () => 'MasqueradeBar');

jest.mock('./ConfirmEmailBanner', () => 'ConfirmEmailBanner');
jest.mock('./AuthenticatedUserDropdown', () => 'AuthenticatedUserDropdown');
jest.mock('./GreetingBanner', () => 'GreetingBanner');

describe('LearnerDashboardHeader', () => {
  describe('snapshots', () => {
    test('with collapsed', () => {
      useIsCollapsed.mockReturnValueOnce(true);
      const wrapper = shallow(<LearnerDashboardHeader />);
      expect(wrapper).toMatchSnapshot();
    });
    test('without collapsed', () => {
      useIsCollapsed.mockReturnValueOnce(false);
      const wrapper = shallow(<LearnerDashboardHeader />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('UserMenu', () => {
    describe('snapshots', () => {
      test('with authenticated user', () => {
        const wrapper = shallow(<UserMenu />);
        expect(wrapper).toMatchSnapshot();
      });
      test('without authenticated user', () => {
        AppContext.authenticatedUser = null;
        const wrapper = shallow(<UserMenu />);
        expect(wrapper).toMatchSnapshot();
      });
    });
  });
});
