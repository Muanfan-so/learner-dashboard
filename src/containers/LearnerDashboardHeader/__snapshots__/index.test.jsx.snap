// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LearnerDashboardHeader UserMenu snapshots with authenticated user 1`] = `
<AuthenticatedUserDropdown
  username="test-username"
/>
`;

exports[`LearnerDashboardHeader UserMenu snapshots without authenticated user 1`] = `""`;

exports[`LearnerDashboardHeader snapshots with collapsed 1`] = `
<Fragment>
  <ConfirmEmailBanner />
  <div
    className="flex-column bg-primary"
  >
    <header
      className="learner-dashboard-header"
    >
      <div
        className="d-flex"
      >
        <div
          className="flex-grow-1"
        >
          <GreetingBanner
            size="small"
          />
        </div>
        <div
          className="my-auto ml-1"
        >
          <UserMenu />
        </div>
      </div>
    </header>
  </div>
  <MasqueradeBar />
</Fragment>
`;

exports[`LearnerDashboardHeader snapshots without collapsed 1`] = `
<Fragment>
  <ConfirmEmailBanner />
  <div
    className="flex-column bg-primary"
  >
    <Image
      className="d-block w-100"
      src="icon/mock/path"
    />
    <header
      className="learner-dashboard-header"
    >
      <div
        className="d-flex"
      >
        <Button
          as="a"
          href="http://localhost:18000/dashboard/programs"
          iconBefore={[MockFunction icons.Program]}
          variant="inverse-tertiary"
        >
          Switch to Programs
        </Button>
        <div
          className="flex-grow-1"
        />
        <UserMenu />
      </div>
    </header>
    <GreetingBanner
      size="large"
    />
  </div>
  <MasqueradeBar />
</Fragment>
`;
